atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -402,8 +402,21 @@ protected void checkGetPreconditions(TableReference tableRef) {
                         Ordering.from(UnsignedBytes.lexicographicalComparator())
                                 .onResultOf(entry -> entry.getKey().getColumnName()),
                         from -> from.getLhSide());
-        // Filter empty columns.
-        return Iterators.filter(mergedIterator, entry -> entry.getValue().length > 0);
+
+        return filterDeletedValues(mergedIterator, tableRef);
+    }
+
+    private Iterator<Map.Entry<Cell, byte[]>> filterDeletedValues(
+            Iterator<Map.Entry<Cell, byte[]>> unfiltered,
+            TableReference tableReference) {
+        Meter emptyValueMeter = getMeter(AtlasDbMetricNames.CellFilterMetrics.EMPTY_VALUE, tableReference);
```

Comment:
do you want to track global empty value filtering? or per transaction?

Reply:
per table should be fine (at least for now)